site/book/07-effective-customizations/00.md

```diff
@@ -0,0 +1,17 @@
+Kubernetes configuration packages and customizations go hand in hand, all the 
+packaging tools enable package customization. In this chapter we cover effective 
+customizations techniques the kpt hydration and packaging enables.  We show how
+providing customization through parameters has some pitfalls and recommend
+alternatives where the contents of the package are no hidden behind a facade.
+Some of these alternatives are only possible because kpt has made an investment 
+into bulk editing with KRM functions and upstream merging. 
```

Comment:
link to https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md